### PR TITLE
Make NoteRef and MyNoteMarker clickable

### PR DESCRIPTION
**Pull request by JJK96**
Click on the MyNotes image or NoteRef to go to the corresponding activity

It works by wrapping the reference text in `<a> </a>` with a special url format, which is parsed by the UriAnalyzer to open the right activity

### CODE REVIEW

File: app/src/main/java/net/bible/android/control/link/LinkControl.kt [hunk 1]
```diff
@@ -255,6 +262,26 @@ class LinkControl @Inject constructor(
...
+    private fun showMyNote(osisRef: String) {
+        val activity: MainBibleActivity = CurrentActivityHolder.getInstance().currentActivity as MainBibleActivity
+        val osisParser = OsisParser()
+        val verseRange = osisParser.parseOsisRef(this.currentPageManager.currentBible.versification, osisRef)
+        activity.fullScreen = false
+        currentPageManager.showMyNote(verseRange)
+        activity.invalidateOptionsMenu()
+        activity.documentViewManager.buildView()
+    }
+
```
Comment (JJK96) [inline app/src/main/java/net/bible/android/control/link/LinkControl.kt:284]: Currently there's a bit of code duplication between these functions and https://github.com/AndBible/and-bible/blob/13893812e8e7839de5b682f1370dddac8a13082b/app/src/main/java/net/bible/android/view/activity/page/actionmode/VerseMenuCommandHandler.java#L67, but I don't know what is a good place to put the common function.

File: app/src/main/java/net/bible/android/control/link/LinkControl.kt [hunk 2]
```diff
@@ -255,6 +262,26 @@ class LinkControl @Inject constructor(
         }
     }
 
+    private fun showNote(osisRef: String?, note: String) {
```
Comment (tuomas2) [inline app/src/main/java/net/bible/android/control/link/LinkControl.kt:265]:
```suggestion
    // TODO: Currently there's a bit of code duplication between these functions and 
    // VerseMenuCommandHandler.java:handleMenuRequest
    private fun showNote(osisRef: String?, note: String) {
```

File: app/src/main/java/net/bible/service/format/osistohtml/taghandler/MyNoteMarker.kt [hunk 3]
```diff
@@ -40,7 +41,12 @@ class MyNoteMarker(private val parameters: OsisToHtmlParameters,
     override fun start(attr: Attributes) {
         if (parameters.isShowMyNotes) {
             if (myNoteVerses.contains(verseInfo.currentVerseNo)) {
-                writer.write("<img src='file:///android_asset/images/pencil16x16.png' class='myNoteImg'/>")
+                var tag: String = "<a href='%s:%s'><img src='%s' class='myNoteImg'/></a>".format(
+                    Constants.MYNOTE_PROTOCOL,
+                    parameters.basisRef.osisID + verseInfo.currentVerseNo,
+                    "file:///android_asset/images/pencil16x16.png"
+                )
```
Comment (tuomas2) [inline app/src/main/java/net/bible/service/format/osistohtml/taghandler/MyNoteMarker.kt:48]: I like more kotlin's string intrepolation syntax

Comment (JJK96) [reply app/src/main/java/net/bible/service/format/osistohtml/taghandler/MyNoteMarker.kt:48]: I'll use that next time then

**Comment by tuomas2** [comment]
Will be included in build 368

**Pull request merged**